### PR TITLE
Handle windows 10 and .NET 4.6 

### PR DESCRIPTION
**Pull request by Annih**
cc: @aboten

### CODE REVIEW

File: attributes/ms_dotnet4.rb [hunk 1]
```diff
@@ -57,7 +57,7 @@
     # Starting with windows 8 and Server 2012 old version of .NET Framework 4 are builtin or included as feature
     if nt_version >= 6.2
       require 'chef/win32/version'
-      if node['kernel']['os_info']['product_type'] != Chef::ReservedNames::Win32::Version::VER_NT_WORKSTATION
+      if node['kernel']['os_info']['product_type'] == Chef::ReservedNames::Win32::Version::VER_NT_WORKSTATION
```
Comment (brugidou) [inline attributes/ms_dotnet4.rb]: It is really not clear what is going on with version in this file. Can we have comments explaining a bit more what is each constant or number used?

File: attributes/ms_dotnet4.rb [hunk 2]
```diff
@@ -71,5 +71,15 @@
       hotfix_id = nt_version == 6.3 ? 'KB2934520' : 'KB2901982'
       default['ms_dotnet']['versions']['4.5.2']['package']['not_if']      = "wmic path Win32_QuickFixEngineering WHERE HotFixID='#{hotfix_id}' | FindStr #{hotfix_id}"
     end
+
+    if nt_version < 10.0
```
Comment (brugidou) [inline attributes/ms_dotnet4.rb]: Comments please. Could we have something a bit more tidy than nested if statements?

**Comment by brugidou** [comment]
Please update the readme too. (BTW there is a mistake on the readme about using feature/package attributes for ms dotnet 4, it is `node['ms_dotnet']['versions'][desired_version]['feature']` and `node['ms_dotnet']['versions'][desired_version]['package']`. Idem for patch.

**Comment by yeungs** [comment]
Could this recipe also include the installation of this security run time patch? It is a pretty well known bug. 

Technical information
http://blogs.msdn.com/b/dotnet/archive/2015/07/28/ryujit-bug-advisory-in-the-net-framework-4-6.aspx 

The installation instructions can be found here.
https://support.microsoft.com/en-us/kb/3086251

**Comment by alexbacchin** [comment]
will this be merged anytime soon or should I fork it ?

**Comment by Annih** [comment]
Hello @alexbacchin 

A new version is coming soon, with many changes on how versions are handled + 4.6 & 4.6.1 support.
We'll not release a new version with only 4.6 support, but if you want we can merge it into master branch, before the biger change.

Regards.

**Comment by albeekj** [comment]
Can this be merged in the mean time till the new stuff has been tested and implemented?

**Comment by albeekj** [comment]
@Annih I noticed there is are conflicts that must be resolved, any chance they can be resolved to have this pushed into the current version?

**Comment by Annih** [comment]
This will be included in release v2.6.2

**Comment by Annih** [comment]
OK version 2.6.2 released :)

**Pull request closed without merging**